### PR TITLE
feat: Handle Grand Hunt differents rewards + Tooltips

### PR DESCRIPTION
**Pull request by Fia3D**
image
image

### CODE REVIEW

File: SavedInstances/Core/Core.lua [hunk 1]
```diff
@@ -2514,6 +2514,43 @@ hoverTooltip.ShowDragonflightRenownTooltip = function (cell, arg, ...)
...
+  end
+
+  local toonstr = (db.Tooltip.ShowServer and toon) or strsplit(' ', toon)
+
+  indicatortip:AddHeader(ClassColorise(t.Class, toonstr), string.format("%d/3", totalDone))
+
+  local naming = {
+    "|cffa335ee" .. "Epic" .. FONT_COLOR_CODE_CLOSE,
+    "|cff42a4f5" .. "Rare" .. FONT_COLOR_CODE_CLOSE,
+    "|cff1eff00" .. "Uncommon" .. FONT_COLOR_CODE_CLOSE,
```
Comment (LiangYuxuan) [inline SavedInstances/Core/Core.lua:2540]: Use global string `ITEM_QUALITY4_DESC`, `ITEM_QUALITY3_DESC`, `ITEM_QUALITY2_DESC` for Epic, Rare and Uncommon. Also you can use `MAW_BUFF_QUALITY_STRING_EPIC`, `MAW_BUFF_QUALITY_STRING_RARE`, `MAW_BUFF_QUALITY_STRING_UNCOMMON` with color codes.

File: SavedInstances/Core/Core.lua [hunk 2]
```diff
@@ -2514,6 +2514,43 @@ hoverTooltip.ShowDragonflightRenownTooltip = function (cell, arg, ...)
...
+  }
+  local IDs = P.TrackedQuest[index].relatedQuest
+
+  for i, questID in ipairs(IDs) do
+    indicatortip:AddLine(
+      naming[i],
+      t.Progress[index][questID] and GREENFONT .. L["Obtained"] .. FONTEND or "Unobtained"
```
Comment (LiangYuxuan) [inline SavedInstances/Core/Core.lua:2547]: Use global string. https://github.com/SavedInstances/SavedInstances/blob/92142fcd6a380cfcaa8c1019e042578afed8983d/SavedInstances/Core/Core.lua#L2412-L2413

File: SavedInstances/Modules/Progress.lua [hunk 3]
```diff
@@ -485,8 +514,15 @@ Module.TrackedQuest = {
...
     weekly = true,
-    quest = 70906,
-    relatedQuest = {70906},
+    func = GrandHuntUpdate,
+    showFunc = GrandHuntShow,
+    resetFunc = GrandHuntReset,
+    relatedQuest = {
+      70906, -- Epic
+      71136, -- Rare
+      71137 -- Uncommon
```
Comment (LiangYuxuan) [inline SavedInstances/Modules/Progress.lua:523]: Tailing comma

**Review by LiangYuxuan — CHANGES_REQUESTED** [review]
Thank you! After soloing global string and little format issue it'll be great.

File: SavedInstances/Core/Core.lua (hunk 1 above)
Comment (Fia3D) [reply SavedInstances/Core/Core.lua]: I ended used MAW_BUFF_QUALITY_STRING but can't seem to pass the static check 

```
    SavedInstances/Core/Core.lua:14:38: accessing undefined variable 'MAW_BUFF_QUALITY_STRING_EPIC'
    SavedInstances/Core/Core.lua:15:38: accessing undefined variable 'MAW_BUFF_QUALITY_STRING_RARE'
    SavedInstances/Core/Core.lua:16:42: accessing undefined variable 'MAW_BUFF_QUALITY_STRING_UNCOMMON'
```

I tried to add on top of the file 
`local MAW_BUFF_QUALITY_STRING_EPIC = MAW_BUFF_QUALITY_STRING_EPIC` etc...
But I still get the error, any idea ?

Comment (Fia3D) [reply SavedInstances/Core/Core.lua]: Thank you, everything should be fixed now

**Review by LiangYuxuan — APPROVED** [review]

**Comment by agentschmitt** [comment]
could we maybe get a setting to just track the epic reward. would be nice for those who dont care about the additional loot :)
if not its also fine as we can see progress with 1/3 then, would just look better with just showing the complete icon :)

**Pull request merged**